java-util/src/main/java/io/druid/java/util/common/granularity/AllGranularity.java

```diff
@@ -38,14 +38,7 @@
   private final DateTime maxDateTime = new DateTime(MAX_INSTANT);
   private final DateTime minDateTime = new DateTime(MIN_INSTANT);
 
-  private static final AllGranularity INSTANCE = new AllGranularity();
-
-  private AllGranularity() {}
-
-  public static AllGranularity getInstance()
-  {
-    return INSTANCE;
-  }
+  public AllGranularity() {}
```

Comment:
Singleton is still a problem or it could be turned back to singleton?

(BTW static singleton instance could be held in `Granularities` or `GranularityType` if needed, not necessarily in `AllGranularity`)